providers/dns/hurricanedns/hurricanedns.go

```diff
@@ -0,0 +1,221 @@
+	}
+
+	resp, err := http.Post(d.heBaseURL, "application/x-www-form-urlencoded", bytes.NewBufferString(form.Encode()))
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 400 {
+		return nil, fmt.Errorf("dns.he.net request failed with HTTP status code %d", resp.StatusCode)
+	}
+
+	body1, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	doc, err := goquery.NewDocumentFromReader(bytes.NewReader(body1))
+	if err != nil {
+		return nil, err
+	}
+
+	if doc.Find("#dns_err").Text() == "Incorrect" {
+		return doc, fmt.Errorf("unable to login to dns.he.net please check username and password")
+	}
+
+	return doc, nil
+}
+
+//Resolve zone ID from HTTP Document
+func (d *DNSProvider) getZoneID(zone string) (string, error) {
+	payload := make(map[string]string)
+
+	doc, err := d.SendRequest(payload)
+	if err != nil {
+		return "", err
+	}
+
+	zones := make(map[string]string)
+	doc.Find("#domains_table tbody td:nth-child(2) img").Each(func(i int, s *goquery.Selection) {
+		domain, _ := s.Attr("name")
+		onclick, _ := s.Attr("onclick")
+		r, _ := regexp.Compile("hosted_dns_zoneid=(\\d+)")
+		id := r.FindStringSubmatch(onclick)
+		zones[domain+"."] = id[1]
+	})
+
+	if val, ok := zones[zone]; ok {
+		if len(val) == 0 {
+			return "", fmt.Errorf("zone %s found, but zone id is not valid", zone)
+		}
+
+		return val, nil
+	}
+
+	return "", fmt.Errorf("zone %s not found", zone)
+}
+
+//Resolve record-set ID from HTTP Document
+func (d *DNSProvider) getRecordSetID(zoneID string, fqdn string) (string, error) {
+	payload := map[string]string{
+		"hosted_dns_zoneid":   zoneID,
+		"menu":                "edit_zone",
+		"hosted_dns_editzone": "",
+	}
+	doc, err := d.SendRequest(payload)
+	if err != nil {
+		return "", err
+	}
+
+	records := make(map[string]string)
+
+	doc.Find(".generictable .dns_tr").Each(func(i int, s *goquery.Selection) {
+		recordId, _ := s.Attr("id")
+		recordFqdn := s.Find("td:nth-child(3)").Text()
+		records[recordFqdn+"."] = recordId
+	})
+
+	if val, ok := records[fqdn]; ok {
+		if len(val) == 0 {
+			return "", fmt.Errorf("record %s found, but record id is not valid", fqdn)
+		}
+
+		return val, nil
+	}
+
+	return "", fmt.Errorf("record %s not found", fqdn)
+}
+
+//Delete the record-set by send form post to dns.hurricanedns.net
+func (d *DNSProvider) deleteRecordSet(zoneID, recordID string) error {
+	payload := map[string]string{
+		"menu":                  "edit_zone",
+		"hosted_dns_zoneid":     zoneID,
+		"hosted_dns_recordid":   recordID,
+		"hosted_dns_editzone":   "1",
+		"hosted_dns_delrecord":  "1",
+		"hosted_dns_delconfirm": "delete",
+	}
+
+	_, err := d.SendRequest(payload)
+
+	return err
+}
+
+// Present creates a TXT record using the specified parameters
+func (d *DNSProvider) Present(domain, token, keyAuth string) error {
+	fqdn, value, ttl := acme.DNS01Record(domain, keyAuth)
+
+	if ttl < 300 {
+		ttl = 300 // 300 is hurricanedns minimum value for ttl
```

Comment:
Do you have some documentation about this limitation?

Reply:
There is no official documentation for that, this is based on the assumption from their web ui at https://dns.he.net